pkg/api/customization/vsphere/listers.go

```diff
@@ -0,0 +1,401 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vapi/library"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func soapLister(ctx context.Context, fieldName string, cc *v1.Secret, dc string) ([]string, error) {
+	finder, err := getSoapFinder(ctx, cc, dc)
+	if err != nil {
+		return nil, err
+	}
+
+	var data []string
+	switch fieldName {
+	case "clusters":
+		data, err = listClusters(ctx, finder) //tbd
+	case "virtual-machines":
+		data, err = listVirtualMachines(ctx, finder, "virtual-machines")
+	case "templates":
+		data, err = listVirtualMachines(ctx, finder, "templates")
+	case "data-stores":
+		data, err = listDataStores(ctx, finder)
+	case "folders":
+		data, err = listFolders(ctx, finder)
+	case "hosts":
+		data, err = listHosts(ctx, finder)
+	case "networks":
+		data, err = listNetworks(ctx, finder)
+	case "data-centers":
+		data, err = listDataCenters(ctx, finder)
+	case "resource-pools":
+		data, err = listResourcePools(ctx, finder)
+	}
+
+	return data, err
+}
+
+func tagsLister(ctx context.Context, fieldName string, cc *v1.Secret, cat string) ([]string, error) {
```

Comment:
There's 2 `...Lister` funcs and then a bunch of `list...` funcs, is there specific reason for that ? Seems like they both have similar functionality but only the `list..` ones are included in other methods.

Reply:
talked it out a bit with @cbron will switch <blah>list to process<blah> and leave list<blah> as they are.